library/std/src/sys/unix/fd.rs

```diff
@@ -28,7 +33,9 @@ const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
 impl FileDesc {
     pub fn new(fd: c_int) -> FileDesc {
-        FileDesc { fd }
+        assert_ne!(fd, -1);
```

Comment:
If you change this to check against `-1i32` instead of just `-1`, then this also statically verifies that `c_int` is indeed a `i32`.
```suggestion
        assert_ne!(fd, -1i32);
```